### ISSUE TITLE
Vulnerability Detector Refactor: Remoted Cluster error detected

### ISSUE THREAD

**Issue by Rebits** [wazuh/wazuh#20790]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 - `dev-14153-vulndet-refactor ` | Wazuh DB | Manager | Packages | Ubuntu 22 |

| Related Issue| 
| --- |
| https://github.com/wazuh/wazuh/issues/20785 |

### Description

During testing iteration specified in https://github.com/wazuh/wazuh/issues/20785, it occurs several warnings in the worker node:

```
2023/12/13 15:07:19 wazuh-remoted: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).
2023/12/13 15:07:20 wazuh-remoted: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).
2023/12/13 15:07:21 wazuh-remoted: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).
2023/12/13 15:07:22 wazuh-remoted: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).
2023/12/13 15:07:23 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:24 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:25 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2023/12/13 15:07:25 wazuh-syscheckd: INFO: FIM sync module started.
2023/12/13 15:07:25 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:25 wazuh-remoted: ERROR: Could not send message through the cluster after '10' attempts.
2023/12/13 15:07:25 wazuh-remoted: ERROR: Agent '007' invalid or empty group assigned.
2023/12/13 15:07:25 wazuh-remoted: ERROR: Error getting group for agent '007'
2023/12/13 15:07:25 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:26 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:27 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2023/12/13 15:07:27 sca: INFO: Security Configuration Assessment scan finished. Duration: 9 seconds.
2023/12/13 15:07:27 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:28 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:29 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:30 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:31 wazuh-remoted: WARNING: Cluster error detected
2023/12/13 15:07:32 wazuh-remoted: WARNING: Cluster error detected
```

This situation was not possible to replicate. Further research is required

### Evidences and Environment

The evidences and the environment used for this testing can be found in https://github.com/wazuh/wazuh/issues/20785

**Comment by Rebits** [comment]
### Steps to replicate

- Deploy a Wazuh cluster of two nodes (worker and master)
- Configure each manager using the provided configurations in this comment
- Start the master node and then the worker node
- After a few seconds the following warnings should appear in the worker node logs:
```

2023/12/15 10:29:44 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:45 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:46 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:47 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:48 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:49 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:50 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:51 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:52 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:53 wazuh-modulesd: WARNING: Cluster error detected
2023/12/15 10:29:53 wazuh-modulesd: ERROR: Could not send message through the cluster after '10' attempts.
2023/12/15 10:29:53 wazuh-modulesd:agent-upgrade: ERROR: (8123): There has been an error executing the request in the tasks manager.
```

- At this point, if we restart one of the nodes, we can check that modulesd and wazuhdb fault:
```
Killing wazuh-clusterd...
wazuh-modulesd: Process 117108 not used by Wazuh, removing...
wazuh-modulesd not running...
Killing wazuh-monitord...
Killing wazuh-logcollector...
Killing wazuh-remoted...
Process wazuh-remoted couldn't be terminated. It will be killed.
Killing wazuh-syscheckd...
Killing wazuh-analysisd...
wazuh-maild not running...
Killing wazuh-execd...
Killing wazuh-db...
Process wazuh-db couldn't be terminated. It will be killed.
Killing wazuh-authd...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
Killing wazuh-apid...
Wazuh v4.8.0 Stopped
Starting Wazuh v4.8.0...
/var/ossec/api/scripts/wazuh-apid.py:306: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
  ssl_context = ssl.SSLContext(protocol=ssl_protocol)
Started wazuh-apid...
Started wazuh-csyslogd...
Started wazuh-dbd...
2023/12/15 10:32:58 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
Started wazuh-integratord...
Started wazuh-agentlessd...

```

**Comment by pereyra-m** [comment]
## RCA

The worker node prints that **WARNING** message because the `master` node has replied this

```
"Error 1121 - Error connecting with socket: Socket 'task' cannot receive connections"
```

The situation can be reproduced with the following steps:
- Configure a cluster
- Stop the worker node
- Kill `wazuh-modulesd` in the master
- Restart the worker node

The worker node will try to send a clustered message to the socket located in `queue/tasks/task`, this one exists but no one is listening. 

This error might be related to #20424 because an instance of `wazuh-modulesd` that wasn't properly closed might be still blocking the socket.

**Comment by Dwordcito** [comment]
Closed, because is duplicated of #20424 based on the RCA.
